### DISCUSSION TITLE
Problem to setup BMP180 (4pin)

### DISCUSSION THREAD

**Discussion by Iboiya** [aaronwmorris/indi-allsky#1322] [Q&A] [answered]
Hi!
I have Opi PC Plus. After system setup, as described in correspound section, I have following errors:
`Jun 14 12:07:20 orangepipcplus [WARNING] MainProcess-1934/Sensor-110 tempSensorBmp180.__init__() [64]: Initializing BMP180 I2C temperature device @ 0x77
Jun 14 12:07:20 orangepipcplus [INFO] MainProcess-1934/MainThread allsky._startSensorWorker() [497]: Starting Sensor-110 worker
Jun 14 12:07:20 orangepipcplus [ERROR] MainProcess-1934/MainThread allsky._startSensorWorker() [490]: Sensor worker exception: 
Jun 14 12:07:20 orangepipcplus [ERROR] MainProcess-1934/MainThread allsky._startSensorWorker() [490]: Sensor worker exception: Valid I2C ports: ((0, 11, 12),)
Jun 14 12:07:20 orangepipcplus [ERROR] MainProcess-1934/MainThread allsky._startSensorWorker() [490]: Sensor worker exception: ValueError: No Hardware I2C on (scl,sda)=(11, 12)
Jun 14 12:07:20 orangepipcplus [ERROR] MainProcess-1934/MainThread allsky._startSensorWorker() [490]: Sensor worker exception:     raise ValueError(
Jun 14 12:07:20 orangepipcplus [ERROR] MainProcess-1934/MainThread allsky._startSensorWorker() [490]: Sensor worker exception:   File "/home/martin/indi-allsky/virtualenv/indi-allsky/lib/python3.9/site-packages/busio.py", line 170, in init
Jun 14 12:07:20 orangepipcplus [ERROR] MainProcess-1934/MainThread allsky._startSensorWorker() [490]: Sensor worker exception:     self.init(scl, sda, frequency)
Jun 14 12:07:20 orangepipcplus [ERROR] MainProcess-1934/MainThread allsky._startSensorWorker() [490]: Sensor worker exception:   File "/home/martin/indi-allsky/virtualenv/indi-allsky/lib/python3.9/site-packages/busio.py", line 36, in __init__
Jun 14 12:07:20 orangepipcplus [ERROR] MainProcess-1934/MainThread allsky._startSensorWorker() [490]: Sensor worker exception:     return busio.I2C(SCL, SDA)
Jun 14 12:07:20 orangepipcplus [ERROR] MainProcess-1934/MainThread allsky._startSensorWorker() [490]: Sensor worker exception:   File "/home/martin/indi-allsky/virtualenv/indi-allsky/lib/python3.9/site-packages/board.py", line 444, in I2C
Jun 14 12:07:20 orangepipcplus [ERROR] MainProcess-1934/MainThread allsky._startSensorWorker() [490]: Sensor worker exception:     i2c = board.I2C()
Jun 14 12:07:20 orangepipcplus [ERROR] MainProcess-1934/MainThread allsky._startSensorWorker() [490]: Sensor worker exception:   File "/home/martin/indi-allsky/indi_allsky/devices/sensors/tempSensorBmp180.py", line 65, in __init__
Jun 14 12:07:20 orangepipcplus [ERROR] MainProcess-1934/MainThread allsky._startSensorWorker() [490]: Sensor worker exception:     self.sensors[0] = a_sensor(self.config, pin_1_name=a_sensor_pin_1_name, i2c_address=a_sensor_i2c_address)
Jun 14 12:07:20 orangepipcplus [ERROR] MainProcess-1934/MainThread allsky._startSensorWorker() [490]: Sensor worker exception:   File "/home/martin/indi-allsky/indi_allsky/sensor.py", line 283, in init_sensors
Jun 14 12:07:20 orangepipcplus [ERROR] MainProcess-1934/MainThread allsky._startSensorWorker() [490]: Sensor worker exception:     self.init_sensors()  # sensors before dew heater and fan
Jun 14 12:07:20 orangepipcplus [ERROR] MainProcess-1934/MainThread allsky._startSensorWorker() [490]: Sensor worker exception:   File "/home/martin/indi-allsky/indi_allsky/sensor.py", line 103, in saferun
Jun 14 12:07:20 orangepipcplus [ERROR] MainProcess-1934/MainThread allsky._startSensorWorker() [490]: Sensor worker exception:     self.saferun()`

`"TEMP_SENSOR": {
    "A_CLASSNAME": "blinka_temp_sensor_bmp180_i2c",
    "A_PIN_1": "PA12",
    "A_USER_VAR_SLOT": 10,
    "A_I2C_ADDRESS": "0x77",`

I can't figure out what the problem is. Help me please.

**Comment by aaronwmorris** [comment]
You may just need to enable the I2C overlay for your board.  Try the following, assuming you are using armbian:

Run "sudo armbian-config", select "System (System and security settings)" then "Hardware (Toggle hardware configuration)", and mark i2c0 for installation. Select save and reboot your Orange Pi

**Reply by Iboiya** [reply]
Thank’s, I will check it.

**Reply by aaronwmorris** [reply]
@Iboiya Did enabling the I2C overlay work for you?

**Reply by Iboiya** [reply]
Yes! And another problem appeared:
`orangepipcplus [ERROR] MainProcess-1940/MainThread allsky._startSensorWorker() [490]: Sensor worker exception: PermissionError: [Errno 13] Permission denied: '/dev/i2c-0'`

GPIO Permissions doesn’t work, I have already check.

**Reply by aaronwmorris** [reply, marked as answer]
I'll work on the setup script for this issue... try the following:

```
sudo usermod -a -G i2c $(whoami)
```

And reboot.

**Reply by Iboiya** [reply]
It works now, thanks!

**Comment by aaronwmorris** [comment]
I have merged #1341 to fix the i2c group membership problem automatically.
